src/Transaction/TransactionBuilder.php

```diff
@@ -265,10 +265,11 @@ public function getTransaction()
                 $script = $this->transaction->getInputs()->getInput($i)->getScript();
             }
 
-            $transaction->getInputs()->getInput($i)->setScript($script);
+            $input = $transaction->getInputs()->getInput($i);
+            $transaction->getInputs()->setInput($i, new TransactionInput($input->getTransactionId(), $input->getVout(), $script, $input->getSequence()));
```

Comment:
Hmm, this appears to be duplicated in SignatureHash - perhaps add a replace() method to TransactionInputCollection (and maybe outputs too) ?